File: lib/services/retour_user_service.dart
import 'package:cloud_firestore/cloud_firestore.dart';
import 'package:viro_team_v2/config/project_config.dart';
import 'package:viro_team_v2/constants/firestore_fields.dart';
import 'package:viro_team_v2/utils/firestore_instance.dart';

/// Retours utilisateur (objectifs onboarding, feedback produit, etc.).
class RetourUserService {
  RetourUserService({FirebaseFirestore? firestore})
      : _db = firestore ?? appFirestore;

  final FirebaseFirestore _db;

  static const String typeClubSetupObjectives = 'club_setup_objectives';

  /// Enregistre les objectifs choisis à la création d'un club.
  Future<void> saveClubSetupObjectives({
    required String userId,
    required String clubId,
    required Set<String> objectiveKeys,
    String? clubName,
    String? clubSport,
    String? memberCountRange,
  }) async {
    if (objectiveKeys.isEmpty) return;

    final labels = objectiveKeys.map(ClubObjectives.label).toList();

    await _db.collection(ProjectConfig.retourUserCollection).add({
      FirestoreFields.userId: userId,
      FirestoreFields.clubId: clubId,
      'type': typeClubSetupObjectives,
      if (clubName != null) FirestoreFields.clubName: clubName,
      if (clubSport != null) FirestoreFields.clubSport: clubSport,
      FirestoreFields.objectives: objectiveKeys.toList(),
      FirestoreFields.objectivesLabels: labels,
      if (memberCountRange != null && memberCountRange.isNotEmpty)
        FirestoreFields.memberCountRange: memberCountRange,
      FirestoreFields.createdAt: FieldValue.serverTimestamp(),
    });
  }
}

File: portal/src/lib/clubSetup/retourUserService.ts
import { addDoc, collection, serverTimestamp } from "firebase/firestore";
import { ClubObjectives } from "@/lib/clubSetup/constants";
import { getAppFirestore } from "@/lib/firebase/app";
import { Collections, Fields } from "@/lib/firebase/constants";

export const RETOUR_USER_TYPE_CLUB_SETUP_OBJECTIVES = "club_setup_objectives";

/** Enregistre les objectifs choisis à la création d’un club. */
export async function saveClubSetupObjectives(params: {
  userId: string;
  clubId: string;
  objectiveKeys: Set<string>;
  clubName?: string;
  clubSport?: string;
  memberCountRange?: string | null;
}): Promise<void> {
  if (params.objectiveKeys.size === 0) return;

  const labels = [...params.objectiveKeys].map((key) => ClubObjectives.label(key));

  await addDoc(collection(getAppFirestore(), Collections.retourUser), {
    [Fields.userId]: params.userId,
    [Fields.clubId]: params.clubId,
    [Fields.type]: RETOUR_USER_TYPE_CLUB_SETUP_OBJECTIVES,
    ...(params.clubName ? { [Fields.clubName]: params.clubName } : {}),
    ...(params.clubSport ? { [Fields.clubSport]: params.clubSport } : {}),
    [Fields.objectives]: [...params.objectiveKeys],
    [Fields.objectivesLabels]: labels,
    ...(params.memberCountRange
      ? { [Fields.memberCountRange]: params.memberCountRange }
      : {}),
    [Fields.createdAt]: serverTimestamp(),
  });
}
